integration/dkg/dkg_whiteboard_test.go

```diff
@@ -0,0 +1,261 @@
+
+	chainID := flow.Testnet
+
+	// we run the DKG protocol with N consensus nodes
+	N := 10
+	conIdentities := unittest.IdentityListFixture(N, unittest.WithRole(flow.RoleConsensus))
+
+	// The EpochSetup event is received at view 100. The phase transitions are
+	// at views 150, 200, and 250. In between phase transitions, the controller
+	// calls the DKG smart-contract every 10 views.
+	//
+	// VIEWS
+	// setup      : 100
+	// polling    : 110 120 130 140 150
+	// Phase1Final: 150
+	// polling    : 160 170 180 190 200
+	// Phase2Final: 200
+	// polling    : 210 220 230 240 250
+	// Phase3Final: 250
+	// final      : 300
+
+	// create and record relevant blocks
+	blocks := make(map[uint64]*flow.Header)
+	var view uint64
+	for view = 100; view <= 250; view += dkgeng.DefaultPollStep {
+		blocks[view] = &flow.Header{View: view}
+	}
+	firstBlock := blocks[100]
+
+	// create the EpochSetup that will trigger the next DKG run with all the
+	// desired parameters
+	epochSetup := flow.EpochSetup{
+		Counter:            999,
+		DKGPhase1FinalView: 150,
+		DKGPhase2FinalView: 200,
+		DKGPhase3FinalView: 250,
+		FinalView:          300,
+		Participants:       conIdentities,
+		RandomSource:       []byte("random bytes for seed"),
+	}
+
+	nodes, _ := createNodes(
+		t,
+		hub,
+		chainID,
+		whiteboard,
+		conIdentities,
+		epochSetup,
+		firstBlock.ID())
+
+	for _, n := range nodes {
+		n.Ready()
+	}
+
+	// trigger the EpochSetupPhaseStarted event for all nodes, effectively
+	// starting the next DKG run
+	for _, n := range nodes {
+		// n.reactorEngine.EpochSetupPhaseStarted(epochSetup.Counter, firstBlock)
+		n.ProtocolEvents.EpochSetupPhaseStarted(epochSetup.Counter, firstBlock)
+	}
+
+	// trigger the BlockFinalized events for each view of interest, effectively
+	// causing the DKG state machine to progress
+	for view = 100; view <= 250; view += dkgeng.DefaultPollStep {
+		time.Sleep(300 * time.Millisecond)
+		hub.DeliverAll()
+		for _, n := range nodes {
+			n.ProtocolEvents.BlockFinalized(blocks[view])
+		}
+	}
+
+	for _, n := range nodes {
+		n.Done()
+	}
+
+	t.Logf("there are %d result(s)", len(whiteboard.results))
+	assert.Equal(t, 1, len(whiteboard.results))
+
+	for _, result := range whiteboard.results {
+		signers := whiteboard.resultSubmitters[flow.MakeID(result)]
+		t.Logf("result %s has %d proposers", flow.MakeID(result), len(signers))
+		assert.Equal(t, N, len(signers))
+	}
+
+	// create and test a threshold signature with the keys computed by dkg
+	sigData := []byte("message to be signed")
+	signers := make([]*signature.ThresholdProvider, 0, N)
+	signatures := []crypto.Signature{}
+	indices := []uint{}
+	for i, n := range nodes {
+		priv, err := n.keyStorage.RetrieveMyDKGPrivateInfo(epochSetup.Counter)
+		require.NoError(t, err)
+
+		signer := signature.NewThresholdProvider("TAG", priv.RandomBeaconPrivKey.PrivateKey)
+		signers = append(signers, signer)
+
+		signature, err := signer.Sign(sigData)
+		require.NoError(t, err)
```

Comment:
Same here, we could verify the signature share against the unique result public key share.